daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_expressions/expressions3.tdml

```diff
@@ -0,0 +1,400 @@
+           then xs:int( ../foo/bar )
+           else xs:int( ../foo/bar + 1 ) )}">
+        <xs:element name="c1" type="xs:string" dfdl:choiceBranchKey="1" />
+        <xs:element name="c2" type="xs:string" dfdl:choiceBranchKey="2" />
+      </xs:choice>
+      </xs:sequence>
+    </xs:group>
+
+    <xs:element name="r">
+      <xs:complexType>
+        <xs:sequence dfdl:separator="|">
+          <xs:element name="e1">
+            <xs:complexType>
+              <xs:sequence>
+                <xs:element name="foo">
+                  <xs:complexType>
+                    <xs:sequence dfdl:separator="|">
+                      <xs:element name="hdr" type="xs:int"/>
+                      <xs:element name="bar" type="xs:float" />
+                      <xs:group ref="ex:g" />
+                    </xs:sequence>
+                  </xs:complexType>
+                </xs:element>
+              </xs:sequence>
+            </xs:complexType>
+          </xs:element>
+          <xs:element name="e2">
+            <xs:complexType>
+              <xs:sequence>
+                <xs:element name="foo">
+                  <xs:complexType>
+                    <xs:sequence dfdl:separator="|">
+                      <xs:element name="hdr" type="xs:int"/>
+                      <xs:element name="bar">
+                        <xs:complexType>
+                          <xs:sequence>
+                            <xs:element name="quux" type="xs:int" />
+                          </xs:sequence>
+                        </xs:complexType>
+                      </xs:element>
+                      <xs:group ref="ex:g" />
+                    </xs:sequence>
+                  </xs:complexType>
+                </xs:element>
+              </xs:sequence>
+            </xs:complexType>
+          </xs:element>
+        </xs:sequence>
+      </xs:complexType>
+    </xs:element>
+ </tdml:defineSchema>
+
+  <tdml:parserTestCase name="test_polymorphic_expr_3" model="s3" root="r"
+    description="Polymorphic path expression">
+    <tdml:document>1|1|a|2|1|b</tdml:document>
+    <tdml:errors>
+      <tdml:error>Schema Definition Error</tdml:error>
+      <tdml:error>cannot be converted to xs:int</tdml:error>
+      <tdml:error>Left operand for operator '+' must have numeric type</tdml:error>
+    </tdml:errors>
+  </tdml:parserTestCase>
+  
+  <tdml:defineSchema name="arraySchema" elementFormDefault="unqualified">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd" />
+    <dfdl:format ref="ex:GeneralFormat" lengthKind="delimited"/>
+
+    <xs:element name="r">
+      <xs:complexType>
+        <xs:sequence dfdl:separator="|">
+          <xs:element name="e1" type="xs:string" minOccurs="3" maxOccurs="3">
+            <xs:annotation>
+              <xs:appinfo source="http://www.ogf.org/dfdl/">
+                <dfdl:assert>{ if (dfdl:occursIndex() eq 1) then fn:true() else ( .[1] eq '1' ) }</dfdl:assert>
```

Comment:
{ . } and { .[1] } are the same thing when the current node is an element, aren't they? I don't think "." is supposed to elevate you to the array containing the current element. I tested this here: https://www.freeformatter.com/xpath-tester.html with the xml:

```
<foo>
  <a>1</a>
  <a>2</a>
  <a>3</a>
</foo>
```

where the expression `foo/a[2]/.[1]` resolved to <a>2</a>

Reply:
Hmmm. There was DFDL workgroup discussion about this a long time back. I will dig in those archives. However, as this is now isolated in a bug ticket about exactly this issue, I will add your comments to that ticket, and not worry about this for this change set.